src/tests.rs

```diff
@@ -527,6 +529,26 @@ mod system_macro {
                         &(Length::new::<meter>((*l).clone())
                             % Length::new::<meter>((*r).clone())).get(meter)))
             }
+
+            #[cfg(feature = "serde")]
+            #[allow(trivial_casts)]
+            fn serde_serialize(v: A<V>) -> bool {
+                let m = Length::new::<meter>((*v).clone());
+                let json_q = serde_json::to_string(&m).expect("Must be able to serialize Quantity");
```

Comment:
Put `let json_q = ...` after `let json_f = ...` to match comparison order below.